### PR TITLE
[BUGFEATURE] - #hashtags fixes + features on Person Details & Donation Details

### PR DESCRIPTION
**Pull request by biscuitehh**
- This PR allows the Hashtag Cells to dynamically size with both width and height (#AccessbilityWin)
- Added an action when tapping the Hashtag Cell on the Person Details Screen

### CODE REVIEW

File: SayTheirNames/Source/Controller/Donations/DonationsMoreDetails/DonationsMoreDetailsController.swift [hunk 1]
```diff
@@ -181,5 +203,21 @@ final class DonationsMoreDetailsController: UIViewController {
...
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        if indexPath.section == DonationSectionLayoutKind.socialMedia.rawValue {
+            let hashtag = donation.hashTags[indexPath.row]
+            
+            if let url = URL(string: hashtag.link) {
+                let vc = SFSafariViewController(url: url)
+                vc.modalPresentationStyle = .overFullScreen
+                self.present(vc, animated: true)
+            }
```
Comment (hybridcattt) [inline SayTheirNames/Source/Controller/Donations/DonationsMoreDetails/DonationsMoreDetailsController.swift:220]: we should open hashtags in the external browser - then they get redirected to the Twitter app

File: SayTheirNames/Source/Utils/Theme/Theme.swift [hunk 2]
```diff
@@ -124,6 +124,9 @@ enum Theme {
                 
             }
         }
-        
+        // Donation Details
+        enum DonationDetails {
+            static let horizontalPadding: CGFloat = 25
```
Comment (hybridcattt) [inline SayTheirNames/Source/Utils/Theme/Theme.swift:129]: should't PersonController have the same padding with this screen?

File: SayTheirNames/Source/View/Hashtag/HashtagView.swift [hunk 3]
```diff
@@ -52,4 +74,9 @@ class HashtagView: UIView {
         self.updateCGColors()
         self.setNeedsDisplay()
     }
+    
+    override var intrinsicContentSize: CGSize {
+        return CGSize(width: self.hashtagLabel.frame.width + (Theme.Components.Padding.medium * 2),
```
Comment (hybridcattt) [inline SayTheirNames/Source/View/Hashtag/HashtagView.swift:79]: shouldn't be `small` horizontally as well?

File: SayTheirNames/Source/Model/Donation.swift [hunk 4]
```diff
@@ -36,16 +36,17 @@ struct Donation: Decodable {
     let type: DonationType?
     let bannerImagePath: String
     let shareable: Shareable
+    let hashTags: [Hashtag]
```
Comment (hybridcattt) [inline SayTheirNames/Source/Model/Donation.swift:39]: "hashtag" goes with a lowercase 't' 😊

File: SayTheirNames/Source/Utils/Theme/Theme.swift (hunk 2 above)
Comment (biscuitehh) [reply SayTheirNames/Source/Utils/Theme/Theme.swift:129]: I think at the time I couldn't count well or the design had different padding - that might be a task for a followup PR

File: SayTheirNames/Source/View/Hashtag/HashtagView.swift [hunk 5]
```diff
@@ -41,8 +58,13 @@ class HashtagView: UIView {
                             leading: leadingAnchor,
                             bottom: bottomAnchor,
                             trailing: trailingAnchor,
-                            padding: UIEdgeInsets(top: 5, left: 5, bottom: 5, right: 5), size: .zero)
+                            padding: UIEdgeInsets(top: Theme.Components.Padding.small,
+                                                  left: Theme.Components.Padding.medium,
```
Comment (hybridcattt) [inline SayTheirNames/Source/View/Hashtag/HashtagView.swift:62]: shouldn't be `small` here too?

File: SayTheirNames/Source/Utils/Theme/Theme.swift (hunk 2 above)
Comment (hybridcattt) [reply SayTheirNames/Source/Utils/Theme/Theme.swift:129]: will you create a card for it or should I do it?

File: SayTheirNames/Source/Controller/Donations/DonationsMoreDetails/DonationsMoreDetailsController+Layout.swift [hunk 6]
```diff
@@ -27,7 +27,7 @@ import UIKit
 // MARK: - UICollectionViewLayout
 extension DonationsMoreDetailsController {
     static var donationMoreDetailsCVLayout: UICollectionViewLayout {
-        let horizontalPadding: CGFloat = 27
+        let horizontalPadding: CGFloat = 25
```
Comment (ohwhen) [inline SayTheirNames/Source/Controller/Donations/DonationsMoreDetails/DonationsMoreDetailsController+Layout.swift:30]: Can we put this in `Theme` please?

File: SayTheirNames/Source/Controller/Donations/DonationsMoreDetails/DonationsMoreDetailsController+Layout.swift [hunk 7]
```diff
@@ -159,9 +71,132 @@ extension DonationsMoreDetailsController {
         }
         
         let config = UICollectionViewCompositionalLayoutConfiguration()
-        config.interSectionSpacing = 28
+        config.interSectionSpacing = 18
```
Comment (ohwhen) [inline SayTheirNames/Source/Controller/Donations/DonationsMoreDetails/DonationsMoreDetailsController+Layout.swift:74]: Same (use `Theme`) here

**Comment by biscuitehh** [comment]
> Great job...left a few comments. The fact that it's a table cell that houses a collectionview threw me off for a sec but overall really nice work 👍

I think long term moving everything to the compositional collectionview layout will be the move

**Pull request merged**